src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -5472,206 +5401,277 @@ address generate_avx_ghash_processBlocks() {
-    __ kmovql(k2, k3);
-    __ evpgatherdd(xmm10, k2, Address(r11, xmm6, Address::times_4, 0), Assembler::AVX_512bit);
-    __ kmovql(k2, k3);
-    __ evpgatherdd(xmm11, k2, Address(r11, xmm7, Address::times_4, 0), Assembler::AVX_512bit);
-
-    //Down convert dword to byte. Final output is 16*6 = 96 bytes long
-    __ evpmovdb(Address(dest, dp, Address::times_1, 0), xmm4, Assembler::AVX_512bit);
-    __ evpmovdb(Address(dest, dp, Address::times_1, 16), xmm5, Assembler::AVX_512bit);
-    __ evpmovdb(Address(dest, dp, Address::times_1, 32), xmm8, Assembler::AVX_512bit);
-    __ evpmovdb(Address(dest, dp, Address::times_1, 48), xmm9, Assembler::AVX_512bit);
-    __ evpmovdb(Address(dest, dp, Address::times_1, 64), xmm10, Assembler::AVX_512bit);
-    __ evpmovdb(Address(dest, dp, Address::times_1, 80), xmm11, Assembler::AVX_512bit);
-
-    __ addq(dest, 96);
-    __ addq(source, 72);
-    __ subq(length, 72);
-    __ jmp(L_process80);
-
-    // Vector Base64 implementation generating 32 bytes of encoded data
-    __ BIND(L_process32);
-    __ cmpl(length, 32);
-    __ jcc(Assembler::below, L_process3);
-    __ evmovdquq(xmm0, Address(source, start_offset), Assembler::AVX_256bit);
-    __ vpermq(xmm0, xmm0, 148, Assembler::AVX_256bit);
-    __ vpshufb(xmm6, xmm0, xmm12, Assembler::AVX_256bit);
-    __ vpmovzxbw(xmm6, xmm6, Assembler::AVX_512bit);
-    __ evpsrlvw(xmm2, xmm6, xmm13, Assembler::AVX_512bit);
-    __ evpsllvw(xmm3, xmm6, xmm14, Assembler::AVX_512bit);
-
-    __ vpsrlq(xmm2, xmm2, 8, Assembler::AVX_512bit);
-    __ vpsllq(xmm3, xmm3, 8, Assembler::AVX_512bit);
-    __ vpandq(xmm3, xmm3, xmm15, Assembler::AVX_512bit);
-    __ vporq(xmm1, xmm2, xmm3, Assembler::AVX_512bit);
-    __ vpsrlq(xmm1, xmm1, 8, Assembler::AVX_512bit);
-    __ vextracti64x4(xmm9, xmm1, 0);
-    __ vpmovzxwd(xmm6, xmm9, Assembler::AVX_512bit);
-    __ vextracti64x4(xmm9, xmm1, 1);
-    __ vpmovzxwd(xmm5, xmm9,  Assembler::AVX_512bit);
-    __ kmovql(k2, k3);
-    __ evpgatherdd(xmm8, k2, Address(r11, xmm6, Address::times_4, 0), Assembler::AVX_512bit);
-    __ kmovql(k2, k3);
-    __ evpgatherdd(xmm10, k2, Address(r11, xmm5, Address::times_4, 0), Assembler::AVX_512bit);
-    __ evpmovdb(Address(dest, dp, Address::times_1, 0), xmm8, Assembler::AVX_512bit);
-    __ evpmovdb(Address(dest, dp, Address::times_1, 16), xmm10, Assembler::AVX_512bit);
-    __ subq(length, 24);
-    __ addq(dest, 32);
-    __ addq(source, 24);
-    __ jmp(L_process32);
-
-    // Scalar data processing takes 3 bytes at a time and produces 4 bytes of encoded data
-    /* This code corresponds to the scalar version of the following snippet in Base64.java
-    ** int bits = (src[sp0++] & 0xff) << 16 |(src[sp0++] & 0xff) << 8 |(src[sp0++] & 0xff);
-    ** dst[dp0++] = (byte)base64[(bits >> > 18) & 0x3f];
-    ** dst[dp0++] = (byte)base64[(bits >> > 12) & 0x3f];
-    ** dst[dp0++] = (byte)base64[(bits >> > 6) & 0x3f];
-    ** dst[dp0++] = (byte)base64[bits & 0x3f];*/
     __ BIND(L_process3);
```

Comment:
Since scalar operation hasnt changed much, could you add new comments or keep the older ones if they are relevant?

Reply:
Added.